catalog/harvest/catalog-harvest-api/src/main/java/org/codice/ddf/catalog/harvest/StorageAdaptor.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.harvest;
+
+/**
+ * A {@code StorageAdaptor} is responsible for persisting {@link HarvestedResource}s to a data
+ * store.
+ *
+ * <p><b> This code is experimental. While this class is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ */
+public interface StorageAdaptor {
```

Comment:
Remind me why we need this interface. It seems like a Listener implementation could be a storage adapter without a need for this interface but it has been a while since I looked over this design.

Reply:
This piece was to abstract away what was done with the harvested resource. For example, whether or not we store just the metadata of a product, or if we store the product as well as its metadata. The Listener could do that, however it would require special logic to decide what to do based on the configuration.

Reply:
There was talk about hiding this interface inside the Listener implementation's module since there will initially only be one implementation.

Reply:
I favor keeping our APIs explicit while we iterate and deleting them when appropriate, for what it's worth.

Reply:
Based on our discussion with @pklinef, I think we should remove this interface from the API for now.
If we do, we may want to remove the `HarvestException` class from this API bundle as well.